app/src/main/java/com/vultisig/wallet/ui/utils/NetworkUtils.kt

```
@@ -29,24 +30,58 @@ object NetworkUtils {
+                } catch (e: Exception) {
+                    Timber.e(e, "Error in onLost")
+                }
             }
 
             override fun onUnavailable() {
-                trySend(false)
+                // Network request could not be fulfilled
+                // Don't modify state, just emit current state
+                try {
+                    updateAndEmit()
+                } catch (t: Throwable) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)